### PR TITLE
swagger, tests, etc

### PR DESCRIPTION
**Pull request by dvoet**
- [ ] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [ ] **Submitter**: If updating admin endpoints, also update [firecloud-admin-cli](https://github.com/broadinstitute/firecloud-admin-cli)
- [ ] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [ ] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [ ] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [ ] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [ ] Tell the tech lead (TL) that the PR exists if they wants to look at it
- [ ] Anoint a lead reviewer (LR). **Assign PR to LR**
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [ ] **TL** sign off
- [ ] **LR** sign off
- [ ] **Product Owner** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Verify swagger UI on dev environment still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via Slack and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

**Comment by dvoet** [comment]
swagger stuff ported from orch

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/DirectoryConfig.scala [hunk 1]
```diff
@@ -1,7 +0,0 @@
-package org.broadinstitute.dsde.workbench.sam.directory
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/directory/DirectoryConfig.scala:1]: config stuff moved to config package

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala [hunk 2]
```diff
@@ -35,22 +37,28 @@ class ResourceService(val accessPolicyDAO: AccessPolicyDAO, val directoryDAO: Jn
   }
 
   def createResource(resourceType: ResourceType, resourceName: ResourceName, userInfo: UserInfo): Future[Set[AccessPolicy]] = {
-    Future.traverse(resourceType.roles) { role =>
-      val roleMembers: Set[SamSubject] = role.roleName match {
-        case resourceType.ownerRoleName => Set(userInfo.userId)
-        case _ => Set.empty
+    accessPolicyDAO.listAccessPolicies(resourceType.name, resourceName) flatMap { policies =>
```
Comment (dvoet) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/service/ResourceService.scala:40]: added check to make sure resource does not already exist

**Comment by dvoet** [comment]
also see https://github.com/broadinstitute/firecloud-develop/pull/544

**Review by MatthewBemis — APPROVED** [review]
Code looks good to me. I was able to test it out using orch's client id and it all worked great. I'm assuming you'll be adding that value to vault as part of https://github.com/broadinstitute/firecloud-develop/pull/544/files right?

**Comment by dvoet** [comment]
it gets the client id out of an existing file

**Comment by MatthewBemis** [comment]
Oh right- `secret/dsde/firecloud/{env}/common/refresh-token-service-account.json`

That's what I get for trying to manually populate the config. Never mind then!

File: src/main/scala/org/broadinstitute/dsde/workbench/sam/config/package.scala [hunk 3]
```diff
@@ -1,13 +1,20 @@
...
 
-import net.ceedubs.ficus.Ficus._
 import net.ceedubs.ficus.readers.ValueReader
 import org.broadinstitute.dsde.workbench.sam.model._
+import net.ceedubs.ficus.Ficus._
+import org.broadinstitute.dsde.workbench.sam.config.DirectoryConfig
 
 /**
-  * Created by mbemis on 6/2/17.
+  * Created by dvoet on 7/18/17.
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/workbench/sam/config/package.scala:9]: @MatthewBemis got owned

Comment (MatthewBemis) [reply src/main/scala/org/broadinstitute/dsde/workbench/sam/config/package.scala:9]: 🤔

**Pull request merged**